common/include/pcl/common/impl/io.hpp

```diff
@@ -51,13 +51,7 @@ pcl::getFieldIndex (const pcl::PointCloud<PointT> &,
                     const std::string &field_name, 
```

Comment:
This function is a good candidate for deprecation.

Reply:
I can see a usecase when `auto` is used, but usually, I've seen `PointT` being a separate template parameter. So it can definitely be deprecated (especially since `decltype(cloud)::PointType` is available)